go/obscuronode/host/client.go

```diff
@@ -72,17 +73,20 @@ func (c *EnclaveRPCClient) GenerateSecret() obscurocommon.EncryptedSharedEnclave
 	return response.EncryptedSharedEnclaveSecret
 }
 
-func (c *EnclaveRPCClient) FetchSecret(report obscurocommon.AttestationReport) obscurocommon.EncryptedSharedEnclaveSecret {
+func (c *EnclaveRPCClient) FetchSecret(report obscurocommon.AttestationReport) (obscurocommon.EncryptedSharedEnclaveSecret, error) {
 	timeoutCtx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 
 	attestationReportMsg := rpc.ToAttestationReportMsg(report)
 	request := generated.FetchSecretRequest{AttestationReportMsg: &attestationReportMsg}
 	response, err := c.protoClient.FetchSecret(timeoutCtx, &request)
 	if err != nil {
-		panic(fmt.Errorf(">   Agg%d: Failed to fetch secret: %w", obscurocommon.ShortAddress(c.nodeID), err))
+		return nil, err
```

Comment:
Do we want to remove the panic here ?

Mind you I prefer it, just making sure because we're still panicking on the rest of the client.

Reply:
I felt that it was legit to remove it because it's not really a disaster, it could definitely happen in prod.

This method is now called ShareSecret, it's one node responding to the request, if it was a bogus request it should fail and be silently dropped. (The requester of the secret would be in a catastrophic state if it doesn't eventually get a response, think that just blocks forever in that case, waiting for some node to have time to help it out.)